src/Mixin.php

```diff
@@ -0,0 +1,539 @@
+     * @throws InvalidArgumentException
+     */
+    public static function nullOrIsAOf($value, $class, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-template ExpectedType of object
+     * @psalm-param class-string<ExpectedType> $class
+     * @psalm-assert iterable<ExpectedType|class-string<ExpectedType>> $value
+     * @param object|string $value
+     * @param string $class
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allIsAOf($value, $class, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert null|empty $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function nullOrIsEmpty($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert iterable<empty> $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allIsEmpty($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert iterable<null> $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allNull($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert null|true $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function nullOrTrue($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert iterable<true> $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allTrue($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert null|false $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function nullOrFalse($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert iterable<false> $value
+     * @param mixed $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allFalse($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert null|lowercase-string $value
+     * @param string $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function nullOrLower($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-assert iterable<lowercase-string> $value
+     * @param string $value
+     * @param string $message
+     * @throws InvalidArgumentException
+     */
+    public static function allLower($value, $message = '');
+
+    /**
+     * @psalm-pure
+     * @psalm-template ExpectedType of object
+     * @psalm-param class-string<ExpectedType> $class
+     * @psalm-assert null|class-string<ExpectedType>|ExpectedType $value
```

Comment:
looks wrong: is the `Assert` implementation also allowing to assert objects here? (if so, feel free to dismiss)

Reply:
Looks okay to me: `nullOrSubclassOf` means - the value is either `null` or a subclass of a specified type.
Or I'm missing something...

Reply:
yep, checked upstream method: it allows objects too 🤷‍♀️